TACC/tup-ui
# feat(tup-cms): tup-348 portal nav

## Overview

Support Portal nav in TUP. _Basically, mirror [Core Portal's](https://github.com/TACC/Core-Portal/blob/main/server/portal/templates/includes/nav_portal.raw.html)._

## Related

- [TUP-348](https://jira.tacc.utexas.edu/browse/TUP-348)

## Changes

- newer Core CMS image
    - to let custom app override portal nav markup url
- add custom app for portal nav
    - template mimics [Portal's](https://github.com/TACC/Core-Portal/blob/main/server/portal/templates/includes/nav_portal.raw.html)
    - view passes user
    - overwrite Core CMS url for portal nav markup

## Testing

### Setup

Run Portal & CMS:
1. `npx nx run tup-ui:serve`
2. `cd apps/tup-cms`
3. `docker-compose -f docker-compose.dev.yml build --no-cache`
4. `docker-compose -f docker-compose.dev.yml up --force-recreate`

### Test

1. Load https://localhost:8000.
2. (top right) See "⃝ Log in" link.
3. Click "⃝ Log in" link.
4. See "Log In" form.
5. Log in.
6. (top right) See "&emsp; your_username".
7. Click username.
8. See "&emsp; My Dashboard" and "&emsp; Log Out" links.
9. Log out.
10. Verify log out.
    - Refresh page to refresh state of server app.

**Known Issues:**
<sup>4. Form has font issues. No ticket.
4. Form has no responsiveness. No ticket.
6. 8. No icons when user is logged in. No ticket.
10. Initial log out does not log user out of CMS. No ticket.
10. Initial log out does not update header. ([Why not?](https://stackoverflow.com/a/59955142/11817077)) No ticket.</sup>

## UI

| step<br >_status_<br ><sub>known issue</sub> | image |
| - | - |
| 1.&nbsp;2.<br >_logged&nbsp;**out**_ | ![0 logged out](https://user-images.githubusercontent.com/62723358/197076083-34f13812-b9fd-4c36-9b61-3acbd4771c4d.png)
| 3.&nbsp;4.<br >_logged&nbsp;**out**_ | ![1 log in form](https://user-images.githubusercontent.com/62723358/197076081-f12df05d-525c-4a69-b661-431fd67a2a1c.png)
| 5.&nbsp;6.&nbsp;7.&nbsp;8.&nbsp;9.<br >_logged&nbsp;**in**_ | ![2 logged in](https://user-images.githubusercontent.com/62723358/197076079-a6c7e141-a813-4ca5-805c-7d4400e27b0b.png)
| 10.<br >_logged&nbsp;**out**_<br ><sub>portal&nbsp;nav&nbsp;bug</sub> | ![3a logged out - portal nav bug](https://user-images.githubusercontent.com/62723358/197076076-6b9956f6-86be-4e8d-b111-8f6d39cbe595.png)
| 10.<br >_logged&nbsp;**out**_<br ><sub>cms&nbsp;admin&nbsp;bug</sub> | ![3b logged out - cms admin bug](https://user-images.githubusercontent.com/62723358/197076072-5f260d0d-1ff0-4f3c-a896-bd9bac0b7d5c.png)
